src/execution/helper.rs

```diff
@@ -77,6 +83,14 @@ impl ExecutionHelperWrapper {
             })),
         }
     }
+
+    pub fn get_old_block_number_and_hash(&self) -> Result<(Felt252, Felt252), HintError> {
+        let eh_ref = self.execution_helper.as_ref().borrow();
+        eh_ref.old_block_number_and_hash.ok_or(HintError::AssertionFailed(
+            format!("Block number is probably < {STORED_BLOCK_HASH_BUFFER}.").into_boxed_str(),
```

Comment:
Does this error message make sense?

Reply:
The original hint has a pretty specific error message: `"Inconsistent block number. The constant STORED_BLOCK_HASH_BUFFER is probably out of sync."`

Reply:
Here we just unwrap old_block_number_and_hash. If it is None is it an internal error, which has nothing to do with block number being < stored block number.

Reply:
Yeah seems like I skipped a line here. Will fix it.

Reply:
Nevermind, this is translated directly from `cairo-lang`: 

https://github.com/starkware-libs/cairo-lang/blob/ee7ce74e1159a349d4b77a5f952241b50b1692de/src/starkware/starknet/core/os/syscall_handler.py#L1237